packages/blade/src/components/Carousel/Carousel.native.tsx

```diff
@@ -0,0 +1,211 @@
+      x: slideIndex * boxWidth,
+      y: 0,
+      animated: true,
+    });
+    setActiveSlide(slideIndex);
+  };
+
+  const goToNextSlide = () => {
+    let slideIndex = activeSlide + 1;
+    if (slideIndex >= totalNumberOfSlides) {
+      slideIndex = 0;
+    }
+    goToSlideIndex(slideIndex);
+  };
+
+  const goToPreviousSlide = () => {
+    let slideIndex = activeSlide - 1;
+    if (activeSlide <= 0) {
+      slideIndex = totalNumberOfSlides - 1;
+    }
+    goToSlideIndex(slideIndex);
+  };
+
+  const carouselContext = React.useMemo<CarouselContextProps>(() => {
+    return {
+      visibleItems: _visibleItems,
+      carouselItemWidth,
+      carouselId: undefined,
+      totalNumberOfSlides,
+      boxWidth,
+    };
+  }, [_visibleItems, boxWidth, carouselItemWidth, totalNumberOfSlides]);
+
+  // auto play
+  useInterval(
+    () => {
+      goToNextSlide();
+    },
+    {
+      delay: 6000,
+      // only enable if autoplay is true & user's intent isn't to interact with carousel
+      enable: autoPlay, // TODO
+    },
+  );
+
+  React.useEffect(() => {
+    onChange?.(activeSlide);
+  }, [activeSlide, onChange]);
+
+  React.useEffect(() => {
+    announce(`Slide ${activeSlide + 1} of ${totalNumberOfSlides}`);
+  }, [activeSlide, totalNumberOfSlides]);
+
+  return (
+    <CarouselContext.Provider value={carouselContext}>
+      <BaseBox display="flex" alignItems="center" flexDirection="column">
+        <BaseBox
+          width="100%"
+          position="relative"
+          display="flex"
+          alignItems="center"
+          gap="spacing.4"
+          flexDirection="row"
+        >
+          <ScrollView
+            {...makeAccessible({ label: accessibilityLabel })}
+            ref={containerRef}
+            onLayout={(e) => {
+              setScrollViewWidth(e.nativeEvent.layout.width);
+            }}
+            // Sync active indicator with scroll
+            onMomentumScrollEnd={(e) => {
+              const slideIndex = Math.round(e.nativeEvent.contentOffset.x / boxWidth);
+              setActiveSlide(slideIndex);
+            }}
+            horizontal
+            snapToAlignment="start"
+            decelerationRate="fast"
+            overScrollMode="never"
+            contentInsetAdjustmentBehavior="never"
+            automaticallyAdjustContentInsets={false}
+            showsHorizontalScrollIndicator={false}
+            showsVerticalScrollIndicator={false}
+            snapToInterval={boxWidth}
+            contentInset={{
+              left: 0,
+              right: 0,
+            }}
+            contentOffset={{ x: 0, y: 0 }}
+            // adding some padding so that if a card is placed the shadows don't cut off
+            contentContainerStyle={{ paddingVertical: 10 }}
+          >
+            {React.Children.map(children, (child, index) => {
+              return React.cloneElement(child as React.ReactElement, {
+                index,
+                shouldHaveStartSpacing: index === 0,
```

Comment:
we don't do this on mobile? do we?

Reply:
Nope, mobile is simpler there is only 2 variant and that is visibleItems: 1 or bleed. 

I mean how would you even add Start/End spacing, space hi nehi hai.